### DISCUSSION TITLE
I can't copy Public properties to an external file form CustomAction on WiX

### DISCUSSION THREAD

**Discussion by ahmedhm19** [wixtoolset/issues#6467] [Questions]
I'm trying to copy the value of a public property (which are set from CMD) to an external text file, but it's returning the property name not the value:

```
<Property Id='PROP1'/>
<Property Id='PROP2'/>
<Property Id="CA_DOTNETCA_OnInstall_BeforeInstallingFiles" Value="PROP1=[PROP1];PROP2=[PROP2]" />
<CustomAction Id="CA_DOTNETCA_OnInstall_BeforeInstallingFiles" BinaryKey="BIN_DOTNETCustomActions" DllEntry="CA_OnInstall_BeforeInstallingFiles" Execute="deferred" Return="check" Impersonate="no" />
```

**MyCustomAction :**
-----------------------

  ```
 [CustomAction]
    public static ActionResult CA_OnInstall_BeforeInstallingFiles(Session session)
    {
            CustomActionData data = session.CustomActionData;

            string createText = "text : " + data["PROP1"] + " " + data["PROP2"];
            File.WriteAllText("c:/text.txt", createText);

            return ActionResult.Success;
    }

```
-------------------
And this is the output content of text file :
    
`text : [PROP1] [PROP2]`

**Comment by robmen** [comment]
Yes. What you are doing is manually assigning the string `[PROP1]` to the deferred custom action property `PROP1` and similar for `PROP2`. The custom action is printing out exactly the value you assigned.

It sounds like you actually want the string `[PROP1]` to be _evaluated_ and provide the _value_ of the property `PROP1`. To do that, you must specify when the property should be evaluated. One way to do that:

    <SetProperty Id="CA_DOTNETCA_OnInstall_BeforeInstallingFiles" Value="PROP1=[PROP1];PROP2=[PROP2]"
                 Before="CA_DOTNETCA_OnInstall_BeforeInstallingFiles"  />
